syntax = "proto3";

package fgg.grpc;

message params {
  string name  = 1;
  string value = 2;
};

message keys {
  int32 fromdt = 1;
  int32 todt = 2;
  int32 key = 3;
}

message FggMsg {
  int32 request = 1;
  repeated int32 outkey = 2;
  repeated params values = 3;
  repeated keys relkey = 4;
}

message FggData {
  int32 field = 1;
  oneof value {
    int32  int_value = 2;
    int64  long_value = 3;
    double dbl_value = 4;
    string str_value = 5;
  }
}

service FggDataService {
  rpc requestData(FggMsg) returns (stream FggData);
  rpc persistData(stream FggData) returns (FggMsg);
  rpc queryData(FggMsg) returns (FggMsg);
}
